wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -470,17 +470,14 @@ private[bitcoins] trait TransactionProcessing extends WalletLogger {
       outgoingA = processSpentUtxos(transaction = transaction,
                                     outputsBeingSpent = spentSpendingInfoDbs,
                                     blockHashOpt = blockHashOpt)
-      tuple = DBIO.sequence(Vector(incomingA, outgoingA))
+      updateReceiveStateA = incomingA.flatMap(incoming =>
+        updateUtxoReceiveConfirmedStatesAction(incoming))
+      updateSpentStateA = outgoingA.flatMap(outgoing =>
+        updateUtxoSpentConfirmedStatesAction(outgoing))
+      tuple = DBIO.sequence(Vector(updateReceiveStateA, updateSpentStateA))
       insertResult <- safeDatabase.run(tuple)
       incoming = insertResult.head
       outgoing = insertResult(1)
```

Comment:
you're still doing the `.head` and `(1)` whcih can be fixed with a `.flatMap` as well